pymc/distributions/multivariate.py

```diff
@@ -572,8 +572,15 @@ def moment(rv, size, n, p):
         n = pt.shape_padright(n)
         mode = pt.round(n * p)
         diff = n - pt.sum(mode, axis=-1, keepdims=True)
-        inc_bool_arr = pt.abs(diff) > 0
-        mode = pt.inc_subtensor(mode[inc_bool_arr.nonzero()], diff[inc_bool_arr.nonzero()])
+        if len(mode.shape.eval()) == 1:
```

Comment:
We shouldn't have evals in intermediate graphs. 

I think a proper solution would need an iterative algorithm and I am not sure that's worth the complexity for an edge case of a distribution that won't be properly sampled by our samplers anyway.

User can always pass an initial value manually.

Reply:
Does argsort work with axis?